scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -442,6 +442,9 @@ class FossId internal constructor(
         projectCode: String,
         projectName: String
     ): String {
+        if (url.contains("@")) {
+            log.warn { "The URL should not contain credentials as its interaction with delta scans is unpredictable." }
+        }
```

Comment:
Blank line after this one.

Reply:
@sschuberth  Somehow I cannot reply to the question above...

I wanted to warn the user that it should not give credentials, because FossID scanner will add its own.

`isValidUri` is too generic, it won't allow me to display a specific warning message to this case I want to address.

Same for `replaceCredentialsInUri`: It will silently digest this case, leaving a URL that could be not valid without the given credentials, and won't output a warning making this case hard to debug.